frontend/src/components/Chat/Chat.jsx

```diff
@@ -0,0 +1,123 @@
+import axios from "axios";
+import { Context } from "../../context";
+import { formatDateTime } from "../../utils";
+
+export default function Chat({ socket, user, room, friend }) {
+    const [currentMessage, setCurrentMessage] = useState("");
+    const [messageList, setMessageList] = useState([]);
+    const { setErrorMessage } = useContext(Context);
+
+    useEffect(() => {
+        const fetchMessages = async () => {
+            try {
+                const response = await axios.get(
+                    `${import.meta.env.VITE_HOST}/messages/${friend.user2.id}`,
+                    { withCredentials: true, validateStatus: () => true }
+                );
+    
+                if (response.status === 200) {
+                    setMessageList(response.data.messages);
+                }
+
+                if (response.status === 500) {
+                    setErrorMessage(response.data.error);
+                }
+            } catch (error) {
+                setErrorMessage(`${error.message}: Please try again later.`);
+            }
+            const response = await axios.get(
+                `${import.meta.env.VITE_HOST}/messages/${friend.user2.id}`,
+                { withCredentials: true, validateStatus: () => true }
+            );
+
+            if (response.status === 200) {
+                setMessageList(response.data.messages);
+            }
+        };
+        fetchMessages();
+    }, [room]);
+
+    const sendMessage = async (event) => {
+        event.preventDefault();
+
+        const messageData = {
+            room: room,
+            author: user.username,
+            friendID: friend.user2.id,
+            content: currentMessage
+        };
+
+        try {
+            const response = await axios.post(
+                `${import.meta.env.VITE_HOST}/message`,
+                messageData,
+                { withCredentials: true, validateStatus: () => true }
+            );
+
+            if (response.status === 200) {
+                await socket.emit("send_message", response.data.message);
+                setMessageList((list) => [...list, response.data.message]);
+                setCurrentMessage("");
+            }
+
+            if (response.status === 500) {
+                setErrorMessage(response.data.error);
+            }
+        } catch (error) {
+            setErrorMessage(error);
+        }
+    };
+
+    useEffect(() => {
+        socket.on("receive_message", (data) => {
+            setMessageList((list) => [...list, data]);
+        });
+
+        return () => socket.removeListener("receive_message");
+    }, [socket]);
+
+    return (
+        <div className="chat-section">
+            <ScrollToBottom className="chat-window">
+                {messageList.map((messageContent) => {
+                    return (
+                        <div
+                            key={messageContent.id}
+                            className="message"
+                            id={
```

Comment:
An ID should be unique in the HTML document. This is creating N number of divs with id="you" and M number of divs with id="other"